tools/cd_scripts/e2e_test.sh

```diff
@@ -68,15 +68,77 @@ gcloud storage cp gs://gcsfuse-release-packages/version-detail/details.txt .
 # Writing VM instance name to details.txt (Format: release-test-<os-name>)
 curl http://metadata.google.internal/computeMetadata/v1/instance/name -H "Metadata-Flavor: Google" >>details.txt
 
-# Based on the os type(from vm instance name) in detail.txt, run the following commands to add starterscriptuser
-if grep -q ubuntu details.txt || grep -q debian details.txt; then
-	#  For ubuntu and debian os
-	sudo adduser --ingroup google-sudoers --disabled-password --home=/home/starterscriptuser --gecos "" starterscriptuser
-else
-	#  For rhel and centos
-	sudo adduser -g google-sudoers --home-dir=/home/starterscriptuser starterscriptuser
+# Function to create the local user
+create_user() {
+  if id "${USERNAME}" &>/dev/null; then
+    echo "User ${USERNAME} already exists."
+    return 0
+  fi
+
+  echo "Creating user ${USERNAME}..."
+  if grep -qi -E 'ubuntu|debian' details.txt; then
+    # For Ubuntu and Debian
+    sudo adduser --disabled-password --home "${HOMEDIR}" --gecos "" "${USERNAME}"
+  elif grep -qi -E 'rhel|centos|rocky' details.txt; then
+    # For RHEL, CentOS, Rocky Linux
+    sudo adduser --home-dir "${HOMEDIR}" "${USERNAME}" && sudo passwd -d "${USERNAME}"
+  else
+    echo "Unsupported OS type in details.txt" >&2
+    return 1
+  fi
+  local exit_code=$?
+
+  if [ ${exit_code} -eq 0 ]; then
+    echo "User ${USERNAME} created successfully."
+  else
+    echo "Failed to create user ${USERNAME}." >&2
+  fi
+  return ${exit_code}
+}
+
+# Function to grant sudo access by creating a file in /etc/sudoers.d/
+grant_sudo() {
+  if ! id "${USERNAME}" &>/dev/null; then
+    echo "User ${USERNAME} does not exist. Cannot grant sudo."
+    return 1
+  fi
+
+  SUDOERS_FILE="/etc/sudoers.d/${USERNAME}"
+
+  if sudo test -f "${SUDOERS_FILE}"; then
+    echo "Sudoers file ${SUDOERS_FILE} already exists."
+  else
+    echo "Granting ${USERNAME} NOPASSWD sudo access..."
+    # Create the sudoers file with the correct content
+    if ! echo "${USERNAME} ALL=(ALL:ALL) NOPASSWD:ALL" | sudo tee "${SUDOERS_FILE}" > /dev/null; then
```

Comment:
To make it a bit less confusing to read

```suggestion
    if ! ( echo "${USERNAME} ALL=(ALL:ALL) NOPASSWD:ALL" | ( sudo tee "${SUDOERS_FILE}" > /dev/null ) ); then
```